### PR TITLE
feat: Add simple `shorebird doctor` command that checks for updates

### LINKED ISSUE #152: feat: add `shorebird doctor`
One of our trusted testers expected there to be one after having used it on Flutter. 🤣 

We probably want one anyway over time.

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds a `doctor` command, similar to `flutter doctor`, that checks the current environment for issues. Currently, this only checks to see whether there is a newer version of `shorebird` available.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [x] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/commands/doctor_command.dart [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/command.dart';
+import 'package:shorebird_cli/src/shorebird_version_mixin.dart';
+import 'package:shorebird_cli/src/version.dart';
+
+/// {@template init_command}
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/doctor_command.dart]:
```suggestion
/// {@template doctor_command}
```

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/doctor_command.dart]: Done.

**Review by felangel — APPROVED** [review]
LGTM ✅ 🎉

File: packages/shorebird_cli/lib/src/commands/doctor_command.dart [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import 'package:shorebird_cli/src/shorebird_version_mixin.dart';
+import 'package:shorebird_cli/src/version.dart';
+
+/// {@template doctor_command}
+///
+/// `shorebird doctor`
+/// A command that checks for potential issues with the current shorebird
+/// environment.
+/// {@endtemplate}
+class DoctorCommand extends ShorebirdCommand with ShorebirdVersionMixin {
```
Comment (felangel) [inline packages/shorebird_cli/lib/src/commands/doctor_command.dart:15]: Minor but might be nice to update the shorebird_cli README to include some docs about this command.

Comment (bryanoltman) [reply packages/shorebird_cli/lib/src/commands/doctor_command.dart:15]: Good call, done

**Pull request merged**